### PR TITLE
DDP-7344: dsm pre-filter

### PR DESCRIPTION
**Pull request by SimoneMaiwald**
## Context

draft PR for adding a pre filter to dsm participant list filter function

### CODE REVIEW

**Comment by SimoneMaiwald** [comment]
@gmakharat @Anzori-Ghurtchumelia the last commit is related to the prefilter the previous ones are only checkstyle fixes and configUtil changes

**Comment by SimoneMaiwald** [comment]
@gmakharat @Anzori-Ghurtchumelia commit [3b3f604](https://github.com/broadinstitute/ddp-study-server/pull/1188/commits/3b3f60432a745e5990c10f41d897d1fcc4d3ec57) and [65c828c](https://github.com/broadinstitute/ddp-study-server/pull/1188/commits/65c828c3bcef72c0cfa33f2ccbb4fdebb325225d) are related to the pre-filter feature.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/filter/participant/EmptyFilterParticipantList.java [hunk 1]
```diff
@@ -1,32 +1,14 @@
...
         prepareNecessaryData(queryParamsMap);
-        String realm = queryParamsMap.get(RoutePath.REALM).value();
-        DDPInstanceDto ddpInstanceByGuid = new DDPInstanceDao().getDDPInstanceByInstanceName(realm).orElseThrow();
-        ParticipantWrapperPayload participantWrapperPayload = new ParticipantWrapperPayload.Builder()
-                .withDdpInstanceDto(ddpInstanceByGuid)
-                .withFrom(from)
-                .withTo(to)
-                .build();
-        ElasticSearch elasticSearch = new ElasticSearch();
-        return new ParticipantWrapper(participantWrapperPayload, elasticSearch).getFilteredList();
```
Comment (gmakharat) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/filter/participant/EmptyFilterParticipantList.java]: why this code is deleted?

File: pepper-apis/dsm-core/src/main/resources/liquibase/DDP-7344_preFilter.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="DDP-7344" author="simone">
+        <addColumn tableName="ddp_instance">
```
Comment (gmakharat) [inline pepper-apis/dsm-core/src/main/resources/liquibase/DDP-7344_preFilter.xml:9]: should be here only one column add? as I saw from the code there is something like `queryItems`?

Comment (pegahtah) [reply pepper-apis/dsm-core/src/main/resources/liquibase/DDP-7344_preFilter.xml:9]: Yes I also have the same question

Comment (SimoneMaiwald) [reply pepper-apis/dsm-core/src/main/resources/liquibase/DDP-7344_preFilter.xml:9]: `study_pre_filter` will be the id of the filter from `view_filter` table.
In DDPInstanceDao the sql `SQL_BASE_SELECT` was changed to join ddp_instance and view_filters table and has query_items column from there

**Review by pegahtah — APPROVED** [review]

**Pull request merged**